docs/operators/ocsf/cast.md

```diff
@@ -0,0 +1,90 @@
+---
+title: ocsf::cast
+category: OCSF
+example: 'ocsf::cast'
+---
+
+Casts incoming events to their OCSF type.
+
+```tql
+ocsf::cast [encode_variants=bool, null_fill=bool, timestamp_to_ms=bool]
+```
+
+## Description
+
+The `ocsf::cast` operator casts incoming events to their corresponding OCSF
+event class type. The resulting type is determined by four fields:
+`metadata.version`, `metadata.profiles`, `metadata.extensions` and `class_uid`.
+Events sharing the same values for these fields are cast to the same type.
+Tenzir supports all OCSF versions (including `-dev` versions), all profiles, and
+all event classes. Extensions are currently limited to those versioned with
+OCSF, including the `win` and `linux` extensions.
+
+### `encode_variants = bool`
+
+Setting this option to `true` emits free-form objects such as `unmapped` as JSON
+strings instead of records. Use it when downstream systems expect a JSON
+representation rather than a nested structure.
+
+When `false`, the operator keeps these fields as records, which may lead to
+schema drift between events of the same class if the free-form objects differ.
+Prefer using `unmapped.parse_json()` to project specific fields when possible.
+
+Defaults to `false`.
+
+### `null_fill = bool`
+
+When `true`, the operator back-fills optional fields that do not exist in the
+input with `null` values. This yields schema-complete events for sinks that
+expect a full OCSF record. Leave it at the default `false` to avoid materializing
+unused fields.
+
+### `timestamp_to_ms = bool`
+
+Converts `time` fields into numeric millisecond values.
```

Comment:
That's not the full story, and it doesn't apply to all `time` fields. Here we'd need an explanation on why and where the difference is, and also what the default is.